src/Nethermind/Nethermind.Core/Threading/McsLock.cs

```diff
@@ -175,6 +190,14 @@ private static void SpinTillNextNotNull(ThreadNode node)
         }
 
         private static void SignalUnlock(ThreadNode nextNode)
+        {
+            // Signal the next node to wake up via ThreadPool to return faster
+            // If it was next in line it should be still spinning
+            // rather sleeping so not need to wake up
+            ThreadPool.UnsafeQueueUserWorkItem((node) => BackgroundSignal(node), nextNode, preferLocal: true);
```

Comment:
Is this really faster to offload the signal to the `ThreadPool`?

Reply:
Yes, because to signal requires reaquiring the lock etc; its more expensive than queing to the ThreadPool.

Signalling via the ThreadPool is more lagged; however with this change the next in queue should be spining so not waiting for a signal; is only already sleeping threads that would expericent that latency, but they are already asleep so 🤷‍♂️.

Means a faster return for releasing the lock.